src/daemon.py

```diff
@@ -0,0 +1,88 @@
+import os
+import time
+import pandas as pd
+from web3 import Web3
+from threading import Thread
+from sqlalchemy import create_engine
+from src.imbalances_script import RawTokenImbalances
+from src.config import CHAIN_RPC_ENDPOINTS, CHAIN_SLEEP_TIMES
+
+def get_web3_instance(chain_name):
+    return Web3(Web3.HTTPProvider(CHAIN_RPC_ENDPOINTS[chain_name]))
+
+def get_finalized_block_number(web3):
+    return web3.eth.block_number - 64
+
+def create_db_connection(chain_name):
+    """Helper function that creates a connection to the prod db."""
+    db_name = "mainnet" if chain_name == 'Ethereum' else "xdai"
+    db_url = os.getenv("PROD_DB_URL").format(chain=db_name)
+    
+    return create_engine(f"postgresql+psycopg2://{db_url}")
+
+def fetch_transaction_hashes(chain_name, start_block, end_block):
+    """Fetch transaction hashes beginning start_block."""
+    prod_connection = create_db_connection(chain_name)
```

Comment:
note that we have 2 separate dbs, one for staging and one for prod, so we need to fetch the tx hashes from both.

Reply:
Oh yeah, was just getting from prod to make sure it works. will add staging.